### PR TITLE
update cases for task168, support environment variables in osimage in ventory data and files

### PR DESCRIPTION
**Pull request by tingtli**
Case Name: import_osimage_with_environments_in_yaml
description: update cases for task168, support environment variables in osimage in ventory data and files

UT result
```
xCAT automated test started at Thu Jun 21 02:29:46 2018
******************************
loading Configure file
******************************
Varible:
    CN = mid08tor03cn01
    ISO = /mnt/xcat/iso/redhat/7.5/GA/RHEL-7.5-20180322.0-Server-ppc64le-dvd1.iso
    MN = boston02
******************************
Initialize xCAT test environment by definition in configure file
******************************
******************************
To detect current test environment
******************************
Detecting: OS = linux
Detecting: ARCH = ppc64le
Detecting: HCP = openbmc
******************************
loading test cases
******************************
To run:
import_osimage_with_environments_in_yaml
******************************
Start to run test cases
******************************
------START::import_osimage_with_environments_in_yaml::Time:Thu Jun 21 02:29:47 2018------

RUN:chdef -t node -o mid08tor03cn01 servicenode= monserver=boston02 nfsserver=boston02 tftpserver=boston02  xcatmaster=boston02 [Thu Jun 21 02:29:47 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
1 object definitions have been created or modified.
CHECK:rc == 0   [Pass]

RUN:makedns -n [Thu Jun 21 02:29:48 2018]
ElapsedTime:8 sec
RETURN rc = 0
RUN:if [[ "__GETNODEATTR(mid08tor03cn01,arch)__" = "ppc64" ]] && [[ "__GETNODEATTR(mid08tor03cn01,mgt)__" != "ipmi" ]]; then getmacs -D mid08tor03cn01; fi [Thu Jun 21 02:30:16 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0   [Pass]

RUN:makedhcp -n [Thu Jun 21 02:30:17 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
Warning: [sn02]: xCAT Version mismatch!
         sn02.pok.stglabs.ibm.com: Version 2.14.1 (git commit 4489480054a5b369d789d02c08a0c38f686c7268, built Wed May 30 07:15:51 EDT 2018)
         boston02: Version 2.14.2 (git commit 09f0772835afdc0f962c9e7fe8cef862e9505ad7, built Tue Jun 19 01:58:56 EDT 2018)

Renamed existing dhcp configuration file to  /etc/dhcp/dhcpd.conf.xcatbak

Renamed existing dhcp configuration file to  /etc/dhcp/dhcpd.conf.xcatbak

CHECK:rc == 0   [Pass]

RUN:makedhcp -a [Thu Jun 21 02:30:18 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
Warning: [boston02]: The ip address 172.21.253.104 of node f6pdu15 overlaps with the DHCP dynamic range specified in networks table, will not add this ip address into dhcpd.leases file.
Warning: [boston02]: The ip address 172.21.253.105 of node f6pdu16 overlaps with the DHCP dynamic range specified in networks table, will not add this ip address into dhcpd.leases file.
Warning: [sn02]: xCAT Version mismatch!
         sn02.pok.stglabs.ibm.com: Version 2.14.1 (git commit 4489480054a5b369d789d02c08a0c38f686c7268, built Wed May 30 07:15:51 EDT 2018)
         boston02: Version 2.14.2 (git commit 09f0772835afdc0f962c9e7fe8cef862e9505ad7, built Tue Jun 19 01:58:56 EDT 2018)

Warning: [sn02]: The ip address 172.21.253.104 of node f6pdu15 overlaps with the DHCP dynamic range specified in networks table, will not add this ip address into dhcpd.leases file.

RUN:copycds /mnt/xcat/iso/redhat/7.5/GA/RHEL-7.5-20180322.0-Server-ppc64le-dvd1.iso [Thu Jun 21 02:30:19 2018]
ElapsedTime:31 sec
RETURN rc = 0
OUTPUT:
Copying media to /install/rhels7.5/ppc64le
CHECK:rc == 0   [Pass]

RUN:rootimgdir=`lsdef -t osimage __GETNODEATTR(mid08tor03cn01,os)__-__GETNODEATTR(mid08tor03cn01,arch)__-netboot-compute|grep rootimgdir|awk -F'=' '{print $2}'`; if [ -d $rootimgdir ]; then mv $rootimgdir $rootimgdir.regbak -f;fi [Thu Jun 21 02:30:50 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0   [Pass]

RUN:export GITREPO="/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo";export SWDIR="/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir";xcat-inventory import -f /opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/test.environments.osimage.yaml [Thu Jun 21 02:30:52 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
Importing object: test.environments.osimage
Inventory import successfully!
CHECK:output =~ Importing object: test.environments.osimage     [Pass]
CHECK:output =~ Inventory import successfully!  [Pass]
CHECK:rc == 0   [Pass]

RUN:lsdef -t osimage -o test.environments.osimage [Thu Jun 21 02:30:53 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
Object name: test.environments.osimage
    environvar=GITREPO=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo,SWDIR=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir
    imagetype=linux
    osarch=ppc64le
    osdistroname=rhels7.5-alternate-ppc64le
    osname=Linux
    osvers=rhels7.5-alternate
    otherpkgdir=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir/otherpkgdir/
    otherpkglist=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/otherpkglist/test1.otherpkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/otherpkglist/test2.otherpkglist,
    permission=755
    pkgdir=/install/rhels7.5-alternate/ppc64le,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir/pkgdir/
    pkglist=/opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.pkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/pkglist/test1.pkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/pkglist/test2.pkglist
    postinstall=/opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.postinstall,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/postinstall/test1.postinstall
    profile=compute
    provmethod=netboot
    rootimgdir=/install/custom/test.environments.osimage
    synclists=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/syncfiles/synclist
    usercomment=rhels7.5,test_environment_variables
CHECK:rc == 0   [Pass]

RUN:genimage test.environments.osimage [Thu Jun 21 02:30:54 2018]
ElapsedTime:119 sec
RETURN rc = 0
OUTPUT:
Generating image:
Generating image:
cd /opt/xcat/share/xcat/netboot/rh; ./genimage -a ppc64le -o rhels7.5-alternate -p compute --permission 755 --env GITREPO=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo,SWDIR=/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir --srcdir "/install/rhels7.5-alternate/ppc64le,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir/pkgdir/" --pkglist /opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.pkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/pkglist/test1.pkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/pkglist/test2.pkglist --otherpkgdir "/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/swdir/otherpkgdir/" --otherpkglist /opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/otherpkglist/test1.otherpkglist,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/otherpkglist/test2.otherpkglist, --postinstall /opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.postinstall,/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/postinstall/test1.postinstall --rootimgdir /install/custom/test.environments.osimage --tempfile /tmp/xcat_genimage.122327 test.environments.osimage
111 blocks
/opt/xcat/share/xcat/netboot/rh
111 blocks
/opt/xcat/share/xcat/netboot/rh
 yum -y -c /tmp/genimage.122342.yum.conf --installroot=/install/custom/test.environments.osimage/rootimg/ --disablerepo=* --enablerepo=rhels7.5-alternate-ppc64le-0 --enablerepo=rhels7.5-alternate-ppc64le-1  install  bash nfs-utils openssl dhclient kernel openssh-server openssh-clients wget rsyslog vim-minimal ntp rsyslog rpm rsync ppc64-utils iputils dracut dracut-network e2fsprogs bc file lsvpd irqbalance procps-ng parted net-tools gzip tar xz grub2 grub2-tools bzip2 ethtool dhcp tftp
Package bash-4.2.46-30.el7.ppc64le already installed and latest version
Package 1:nfs-utils-1.3.0-0.54.el7.ppc64le already installed and latest version
Package 1:openssl-1.0.2k-12.el7.ppc64le already installed and latest version
Package 12:dhclient-4.2.5-68.el7.ppc64le already installed and latest version
Package kernel-4.14.0-49.el7a.ppc64le already installed and latest version
Package openssh-server-7.4p1-16.el7.ppc64le already installed and latest version
Package openssh-clients-7.4p1-16.el7.ppc64le already installed and latest version
Package wget-1.14-15.el7_4.1.ppc64le already installed and latest version
Package rsyslog-8.24.0-16.el7.ppc64le already installed and latest version
Package 2:vim-minimal-7.4.160-4.el7.ppc64le already installed and latest version
Package ntp-4.2.6p5-28.el7.ppc64le already installed and latest version
Package rsyslog-8.24.0-16.el7.ppc64le already installed and latest version
Package rpm-4.11.3-32.el7.ppc64le already installed and latest version
Package rsync-3.1.2-4.el7.ppc64le already installed and latest version
Package ppc64-utils-0.14-16.el7.ppc64le already installed and latest version
Package iputils-20160308-10.el7.ppc64le already installed and latest version
Package dracut-033-535.el7.ppc64le already installed and latest version
Package dracut-network-033-535.el7.ppc64le already installed and latest version
Package e2fsprogs-1.42.9-11.el7.ppc64le already installed and latest version
Package bc-1.06.95-13.el7.ppc64le already installed and latest version
Package file-5.11-33.el7.ppc64le already installed and latest version
Package lsvpd-1.7.8-5.el7.ppc64le already installed and latest version
Package 3:irqbalance-1.0.7-11.el7.ppc64le already installed and latest version
Package procps-ng-3.3.10-17.el7.ppc64le already installed and latest version
Cleaning repos: otherpkgs1 rhels7.5-alternate-ppc64le-0
              : rhels7.5-alternate-ppc64le-1
Cleaning up everything
 yum -y -c /tmp/genimage.122342.yum.conf --installroot=/install/custom/test.environments.osimage/rootimg/ --disablerepo=* --enablerepo=rhels7.5-alternate-ppc64le-0 --enablerepo=rhels7.5-alternate-ppc64le-1 --enablerepo=otherpkgs1 install   conserver-xcat perl-Net-Telnet
Package conserver-xcat-8.2.1-1.ppc64le already installed and latest version
Package perl-Net-Telnet-3.03-19.el7.noarch already installed and latest version
Nothing to do
No packages marked for update
Enter the dracut mode. Dracut version: 033. Dracut directory: dracut_033.
Try to load drivers: ext3 ext4 to initrd.
chroot /install/custom/test.environments.osimage/rootimg dracut  -f /tmp/initrd.122342.gz 4.14.0-49.el7a.ppc64le
No '/dev/log' or 'logger' included for syslog logging
Turning off host-only mode: '/sys' is not mounted!
Turning off host-only mode: '/proc' is not mounted!
Turning off host-only mode: '/run' is not mounted!
Turning off host-only mode: '/dev' is not mounted!
the initial ramdisk for stateless is generated successfully.
Try to load drivers: ext3 ext4 to initrd.
chroot /install/custom/test.environments.osimage/rootimg dracut  -f /tmp/initrd.122342.gz 4.14.0-49.el7a.ppc64le
No '/dev/log' or 'logger' included for syslog logging
Turning off host-only mode: '/sys' is not mounted!
Turning off host-only mode: '/proc' is not mounted!
Turning off host-only mode: '/run' is not mounted!
Turning off host-only mode: '/dev' is not mounted!
the initial ramdisk for statelite is generated successfully.
CHECK:rc == 0   [Pass]

RUN:packimage test.environments.osimage [Thu Jun 21 02:32:53 2018]
ElapsedTime:75 sec
RETURN rc = 0
OUTPUT:
Packing contents of /install/custom/test.environments.osimage/rootimg
archive method:cpio
compress method:gzip

CHECK:rc == 0   [Pass]
RUN:rinstall mid08tor03cn01 osimage=test.environments.osimage [Thu Jun 21 02:34:08 2018]
ElapsedTime:24 sec
RETURN rc = 0
OUTPUT:
Provision node(s): mid08tor03cn01
CHECK:rc == 0   [Pass]
CHECK:output =~ Provision node\(s\)\: mid08tor03cn01    [Pass]

RUN:if [[ -f /var/lib/dhcp/db/dhcpd.leases ]]; then cat /var/lib/dhcp/db/dhcpd.leases; elif [[ -f /var/lib/dhcpd/dhcpd.leases ]];then cat /var/lib/dhcpd/dhcpd.leases;elif [[ -f /var/lib/dhcp/dhcpd.leases ]];then cat /var/lib/dhcp/dhcpd.leases; fi [Thu Jun 21 02:34:32 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
# The format of this file is documented in the dhcpd.leases(5) manual page.
# This lease file was written by isc-dhcp-4.2.5

RUN:cat /tmp/mountoutput/file.org [Thu Jun 21 02:34:32 2018]
ElapsedTime:0 sec
RETURN rc = 1
OUTPUT:
cat: /tmp/mountoutput/file.org: No such file or directory

RUN:sleep 300 [Thu Jun 21 02:34:32 2018]
ElapsedTime:300 sec
RETURN rc = 0
OUTPUT:

RUN:a=0;while ! `lsdef -l mid08tor03cn01|grep status|grep booted >/dev/null`; do sleep 10;((a++));if [ $a -gt 60 ];then break;fi done [Thu Jun 21 02:39:32 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:

RUN:ping mid08tor03cn01 -c 3 [Thu Jun 21 02:39:32 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
PING mid08tor03cn01 (172.20.226.1) 56(84) bytes of data.
PING mid08tor03cn01 (172.20.226.1) 56(84) bytes of data.
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=1 ttl=64 time=0.129 ms
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=2 ttl=64 time=0.089 ms
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=3 ttl=64 time=0.066 ms

--- mid08tor03cn01 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2043ms
rtt min/avg/max/mdev = 0.066/0.094/0.129/0.028 ms
CHECK:rc == 0   [Pass]
CHECK:output =~ 64 bytes from mid08tor03cn01    [Pass]

RUN:lsdef -l mid08tor03cn01 | grep status [Thu Jun 21 02:39:34 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
    status=booted
    statustime=06-21-2018 02:37:29
    updatestatus=failed
    updatestatustime=06-14-2018 06:00:52
CHECK:rc == 0   [Pass]
CHECK:output =~ booted  [Pass]

RUN:xdsh mid08tor03cn01 date [Thu Jun 21 02:39:35 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: Thu Jun 21 02:37:29 EDT 2018
CHECK:rc == 0   [Pass]
CHECK:output =~ \d\d:\d\d:\d\d  [Pass]

RUN:xdsh mid08tor03cn01 mount [Thu Jun 21 02:39:36 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: sysfs on /sys type sysfs (rw,relatime)
mid08tor03cn01: proc on /proc type proc (rw,relatime)
mid08tor03cn01: devtmpfs on /dev type devtmpfs (rw,nosuid,size=254567040k,nr_inodes=3977610,mode=755)
mid08tor03cn01: securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
mid08tor03cn01: tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
mid08tor03cn01: devpts on /dev/pts type devpts (rw,relatime,gid=5,mode=620,ptmxmode=000)
mid08tor03cn01: devpts on /dev/pts type devpts (rw,relatime,gid=5,mode=620,ptmxmode=000)
mid08tor03cn01: tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)
mid08tor03cn01: tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
mid08tor03cn01: cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
mid08tor03cn01: pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
mid08tor03cn01: cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
mid08tor03cn01: cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
mid08tor03cn01: cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
mid08tor03cn01: cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
mid08tor03cn01: cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
mid08tor03cn01: cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
mid08tor03cn01: cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
mid08tor03cn01: cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
mid08tor03cn01: cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
mid08tor03cn01: cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
mid08tor03cn01: configfs on /sys/kernel/config type configfs (rw,relatime)
mid08tor03cn01: rootfs on / type tmpfs (rw,relatime,mode=755)
mid08tor03cn01: rw on /.sllocal/log type tmpfs (rw,relatime)
mid08tor03cn01: rpc_pipefs on /var/lib/nfs/rpc_pipefs type rpc_pipefs (rw,relatime)
mid08tor03cn01: systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=25,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=113670)
mid08tor03cn01: mqueue on /dev/mqueue type mqueue (rw,relatime)
mid08tor03cn01: hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,pagesize=2M)
mid08tor03cn01: debugfs on /sys/kernel/debug type debugfs (rw,relatime)
mid08tor03cn01: tmpfs on /run/user/0 type tmpfs (rw,nosuid,nodev,relatime,size=53602944k,mode=700)
CHECK:rc == 0   [Pass]
CHECK:output =~ on / type tmpfs [Pass]

RUN:sleep 120 [Thu Jun 21 02:39:37 2018]
ElapsedTime:120 sec
RETURN rc = 0
OUTPUT:

RUN:ping mid08tor03cn01 -c 3 [Thu Jun 21 02:41:37 2018]
ElapsedTime:2 sec
RETURN rc = 0
OUTPUT:
PING mid08tor03cn01 (172.20.226.1) 56(84) bytes of data.
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=1 ttl=64 time=0.092 ms
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=2 ttl=64 time=0.075 ms
64 bytes from mid08tor03cn01 (172.20.226.1): icmp_seq=3 ttl=64 time=0.066 ms
--- mid08tor03cn01 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 2109ms
rtt min/avg/max/mdev = 0.066/0.077/0.092/0.014 ms
CHECK:rc == 0   [Pass]
CHECK:output =~ 64 bytes from mid08tor03cn01    [Pass]

RUN:rootimgdir=`lsdef -t osimage  __GETNODEATTR(mid08tor03cn01,os)__-__GETNODEATTR(mid08tor03cn01,arch)__-netboot-compute|grep rootimgdir|awk -F'=' '{print $2}'`; if [ -d $rootimgdir.regbak ]; then rm -rf $rootimgdir; mv $rootimgdir.regbak $rootimgdir; fi [Thu Jun 21 02:41:39 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
CHECK:rc == 0   [Pass]

RUN:xdsh mid08tor03cn01  "cat /var/log/xcat/xcat.log" [Thu Jun 21 02:41:40 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: Thu Jun 21 02:35:19 EDT 2018 [info]: xcat.xcatdsklspost: trying to download postscripts...
mid08tor03cn01: Thu Jun 21 02:35:19 EDT 2018 [info]: xcat.xcatdsklspost: postscripts downloaded successfully
mid08tor03cn01: Thu Jun 21 02:35:19 EDT 2018 [info]: xcat.xcatdsklspost: trying to get mypostscript from 172.20.253.31...
mid08tor03cn01: Thu Jun 21 02:35:19 EDT 2018 [info]: xcat.mypostscript: Running postbootscript: syslog
mid08tor03cn01: grep: /etc/rsyslog.d/remote.conf: No such file or directory
mid08tor03cn01: grep: /etc/rsyslog.d/remote.conf: No such file or directory
mid08tor03cn01: Thu Jun 21 02:35:20 EDT 2018 [info]: xcat.mypostscript: postbootscript syslog return with 0
mid08tor03cn01: Thu Jun 21 02:35:20 EDT 2018 [info]: xcat.mypostscript: Running postbootscript: remoteshell
mid08tor03cn01:
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: postbootscript remoteshell return with 0
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: Running postbootscript: syncfiles
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: postbootscript syncfiles return with 0
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: Running postbootscript: otherpkgs
mid08tor03cn01:   Did not install any extra packages.
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: postbootscript otherpkgs return with 0
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [debug]: xcat.mypostscript: node booted successfully, reporting status...
mid08tor03cn01: Thu Jun 21 02:35:22 EDT 2018 [info]: xcat.mypostscript: provision completed.(mid08tor03cn01)

RUN:xdsh mid08tor03cn01  "cat /tmp/synclist" [Thu Jun 21 02:41:41 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: ${GITREPO}/syncfiles/synclist -> /tmp/synclist
CHECK:rc == 0   [Pass]
RUN:xdsh mid08tor03cn01 "cat /tmp/test1.postinstall" [Thu Jun 21 02:41:42 2018]
ElapsedTime:0 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: test postisntall
mid08tor03cn01: test postisntall
mid08tor03cn01: test postisntall
mid08tor03cn01: test postisntall
mid08tor03cn01: test postisntall
CHECK:rc == 0   [Pass]
CHECK:output =~ test postisntall        [Pass]

RUN:xdsh mid08tor03cn01 "cat /tmp/test2.postinstall" [Thu Jun 21 02:41:42 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: test postisntall
CHECK:rc == 0   [Pass]
CHECK:output =~ test postisntall        [Pass]

RUN:xdsh mid08tor03cn01 "rpm -qa |grep -i conserver-xcat" [Thu Jun 21 02:41:43 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: conserver-xcat-8.2.1-1.ppc64le
CHECK:rc == 0   [Pass]

RUN:xdsh mid08tor03cn01 "rpm -qa |grep -i perl-Net-Telnet" [Thu Jun 21 02:41:44 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: perl-Net-Telnet-3.03-19.el7.noarch
CHECK:rc == 0   [Pass]

RUN:xdsh mid08tor03cn01 "rpm -qa |grep -i tftp" [Thu Jun 21 02:41:45 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: tftp-5.2-22.el7.ppc64le
CHECK:rc == 0   [Pass]
RUN:xdsh mid08tor03cn01 "rpm -qa |grep -w dhcp" [Thu Jun 21 02:41:46 2018]
ElapsedTime:1 sec
RETURN rc = 0
OUTPUT:
mid08tor03cn01: dhcp-4.2.5-68.el7.ppc64le
mid08tor03cn01: dhcp-libs-4.2.5-68.el7.ppc64le
mid08tor03cn01: dhcp-common-4.2.5-68.el7.ppc64le
CHECK:rc == 0   [Pass]

------END::import_osimage_with_environments_in_yaml::Passed::Time:Thu Jun 21 02:41:47 2018 ::Duration::720 sec------
------Total: 1 , Failed: 0------

xCAT automated test finished at Thu Jun 21 02:41:47 2018
Please check results in the /opt/xcat/bin/../share/xcat/tools/autotest/result/,
and see /opt/xcat/bin/../share/xcat/tools/autotest/result//failedcases.20180621022946 file for failed cases.
see /opt/xcat/bin/../share/xcat/tools/autotest/result//performance.report.20180621022946 file for time consumption
```

### CODE REVIEW

**Comment by hu-weihua** [comment]
Related task xcat2/xcat2-task-management#168

File: xCAT-test/autotest/testcase/xcat-inventory/templates/environment/gitrepo/syncfiles/synclist [hunk 1]
```diff
@@ -0,0 +1 @@
+/opt/xcat/share/xcat/tools/autotest/testcase/xcat-inventory/templates/environment/gitrepo/syncfiles/synclist -> /tmp/synclist
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/templates/environment/gitrepo/syncfiles/synclist]: should be 
```
${GITREPO}/syncfiles/synclist -> /tmp/synclist
```

File: xCAT-test/autotest/testcase/xcat-inventory/templates/environment/test.environments.osimage.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      arch: ppc64le
+      distribution: rhels7.5-alternate 
+      osdistro: rhels7.5-alternate-ppc64le 
+      osname: Linux
+    filestosync: {{GITREPO}}/syncfiles/synclist
+    deprecated:
+      comments: rhels7.5,test_environment_variables
+    genimgoptions:
+      permission: '755'
+      postinstall: /opt/xcat/share/xcat/netboot/rh/compute.rhels7.ppc64le.postinstall,{{GITREPO}}/postinstall/test1.postinstall,{{GITREPO}}/postinstall/test2.postinstall
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/templates/environment/test.environments.osimage.yaml]: `{{GITREPO}}/postinstall/test2.postinstall` should be invoked in `{{GITREPO}}/postinstall/test1.postinstall`

File: xCAT-test/autotest/testcase/xcat-inventory/templates/environment/gitrepo/postinstall/test1.postinstall [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+#-- Example of booted image versioning
+#-- We want to know, with what configuration (version of the image) each node was booted.
+#-- Hence, we keep image definition files and postscripts in CVS. During image generation we create file /etc/IMGVERSION and fill it with CVS "$Id$" of files with image definition (.pkglist, .exlist, .repolist, .postinstall). Then, during boot, each  "CVS enabled" postscript (see /install/postscripts/cvs_template.sh and /install/postscripts/cvs_template.pl) adds one line to /etc/IMGVERSION. Then you can determine in any time what image you are running and what postscipts in which versions were run.
+#cat /dev/null > $installroot/etc/IMGVERSION
+#for ext in pkglist exlist postinstall repolist; do
+#  [ -r $workdir/$profile.$ext ] && cat $workdir/$profile.$ext | grep -E '^[[:space:]]*#.*[[:space:]]\$Id' >> $installroot/etc/IMGVERSION
+#done
+echo "test postisntall" >> $installroot/tmp/test1.postinstall
+
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/templates/environment/gitrepo/postinstall/test1.postinstall]: should better call `{{GITREPO}}/postinstall/test2.postinstall` here.

File: xCAT-test/autotest/testcase/xcat-inventory/cases.environment [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+cmd:ping $$CN -c 3
+check:rc==0
+check:output=~64 bytes from $$CN
+
+cmd:rootimgdir=`lsdef -t osimage  __GETNODEATTR($$CN,os)__-__GETNODEATTR($$CN,arch)__-netboot-compute|grep rootimgdir|awk -F'=' '{print $2}'`; if [ -d $rootimgdir.regbak ]; then rm -rf $rootimgdir; mv $rootimgdir.regbak $rootimgdir; fi
+check:rc==0
+cmd:xdsh $$CN  "cat /var/log/xcat/xcat.log"
+cmd:xdsh $$CN  "cat /test.synclist"
+check:rc==0
+cmd:xdsh $$CN "cat /tmp/test1.postinstall"
```
Comment (immarvin) [inline xCAT-test/autotest/testcase/xcat-inventory/cases.environment]: should also check the content of the files

**Comment by immarvin** [comment]
i am ok with the change

**Pull request merged**